sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -165,6 +164,17 @@ trait FileFormat {
     }
   }
 
+  /**
+   * Create a file metadata struct column containing fields supported by the given file format.
+   */
+  def createFileMetadataCol(): AttributeReference = {
+    // Strip out the fields' metadata to avoid exposing it to the user. [[FileSourceStrategy]]
+    // avoids confusion by mapping back to [[metadataSchemaFields]].
```

Comment:
Nit .. but in these regular comments, we could just use backticks. `[[...]]` is the syntax for Scaladoc (not for the comments).